IronFoundry.Container.Shared/Utilities/IProcess.cs

```diff
@@ -10,6 +11,7 @@ public interface IProcess : IDisposable
         int ExitCode { get; }
         IntPtr Handle { get; }
         int Id { get; }
+        IReadOnlyDictionary<string, string> Environment { get; }
```

Comment:
Why do we need to expose this?  Does the Warden use this after it's created the process?

This should just be a snapshot of the environment when the process was created, right?  I'm wondering when that would be useful.